src/main/java/org/elasticsearch/index/mapper/core/ByteFieldMapper.java

```diff
@@ -122,24 +130,81 @@ public ByteFieldMapper build(BuilderContext context) {
         }
     }
 
+    public static class ByteFieldType extends NumberFieldType {
```

Comment:
can this one be final? I didn't look at the others but I guess some of them can be too?